### PR TITLE
fix warnings/errors when linking simcards to itilobjects

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #7417

In addition of issue mentionned in #7417, i found global dropdown also failed.
Due to missing `is_helpdesk_visible` and `name` fields

### CODE REVIEW

File: inc/item_ticket.class.php [hunk 1]
```diff
@@ -685,6 +685,10 @@ static function dropdownMyDevices($userID = 0, $entity_restrict = -1, $itemtype
                 && Ticket::isPossibleToAssignType($itemtype)) {
                $itemtable = getTableForItemType($itemtype);
 
+               if (!$DB->fieldExists($itemtable, 'users_id')) {
```
Comment (cedric-anne) [inline inc/item_ticket.class.php:688]: Is it normal to have an item without `users_id` field in `$CFG_GLPI["linkuser_types"]` ?
Maybe problem will be the same for `$CFG_GLPI["linkgroup_types"]`.

File: inc/dropdown.class.php [hunk 2]
```diff
@@ -3258,7 +3258,7 @@ public static function getDropdownFindNum($post, $json = true) {
          $filterHelpdesk = false;
       }
 
-      if ($filterHelpdesk) {
+      if ($filterHelpdesk && $DB->fieldExists($post['table'], 'is_helpdesk_visible')) {
```
Comment (cedric-anne) [inline inc/dropdown.class.php:3261]: If there is no `is_helpdesk_visible` column, then this item should not be in `$CFG_GLPI["helpdesk_visible_types"])`.

File: inc/item_ticket.class.php (hunk 1 above)
Comment (orthagh) [reply inc/item_ticket.class.php:688]: Yes and no.

We have two options for adding items to ticket, adding current user's ones and add a global one.
In the example of simcard, i'm not sure, it's still valid for global part, but anyway, you're right, we should add users_id fields to table to permits also "mine" part (and respect the purpose of the function).

File: inc/dropdown.class.php (hunk 2 above)
Comment (orthagh) [reply inc/dropdown.class.php:3261]: to be removed, will do a global commit for this and #7428

File: install/update_94_95.php [hunk 3]
```diff
@@ -2021,6 +2017,15 @@ function update94to95() {
    }
    // /use_kerberos
 
+   // add missing fields to simcard as they can be associated to tickets
+   if (!$DB->fieldExists(Item_DeviceSimcard::getTable(), 'users_id')) {
+      $migration->addField(Item_DeviceSimcard::getTable(), 'users_id', 'int', [
```
Comment (cedric-anne) [inline install/update_94_95.php:2022]: Maybe adding `groups_id` field could be usefull too.

File: install/mysql/glpi-empty.sql [hunk 4]
```diff
@@ -2240,7 +2242,8 @@ CREATE TABLE IF NOT EXISTS `glpi_items_devicesimcards` (
   KEY `otherserial` (`otherserial`),
   KEY `states_id` (`states_id`),
   KEY `locations_id` (`locations_id`),
-  KEY `lines_id` (`lines_id`)
+  KEY `lines_id` (`lines_id`),
+  KEY `groups_id` (`groups_id`)
```
Comment (cedric-anne) [inline install/mysql/glpi-empty.sql:2247]:
```suggestion
  KEY `users_id` (`users_id`),
  KEY `groups_id` (`groups_id`)
```

File: install/update_94_95.php [hunk 5]
```diff
@@ -2021,6 +2017,23 @@ function update94to95() {
...
+      $migration->addKey(Item_DeviceSimcard::getTable(), 'users_id');
+   }
+   if (!$DB->fieldExists(Item_DeviceSimcard::getTable(), 'groups_id')) {
+      $migration->addField(Item_DeviceSimcard::getTable(), 'groups_id', 'int', [
+         'value' => 0,
+         'after' => 'users_id'
+      ]);
+      $migration->addKey(Item_DeviceSimcard::getTable(), 'groups_id');
+   }
+   // /use_kerberos
```
Comment (cedric-anne) [inline install/update_94_95.php:2035]:
```suggestion
   // /add missing fields to simcard as they can be associated to tickets
```

File: install/mysql/glpi-empty.sql [hunk 6]
```diff
@@ -2431,7 +2435,6 @@ CREATE TABLE `glpi_domains` (
   `users_id_tech` int(11) NOT NULL DEFAULT '0',
   `groups_id_tech` int(11) NOT NULL DEFAULT '0',
   `others` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
-  `is_helpdesk_visible` tinyint(1) NOT NULL DEFAULT '1',
```
Comment (trasher) [inline install/mysql/glpi-empty.sql:2434]: Should be dropped from update if present

Comment (trasher) [reply install/mysql/glpi-empty.sql:2434]: Also, maybe add a note in the PR (and to future release notes :D); to tell users a forced upgrade will be required for this patch to be applied.

**Pull request merged**